src/Psalm/Internal/Analyzer/Statements/Expression/CastAnalyzer.php

```diff
@@ -61,9 +60,8 @@ public static function analyze(
                     }
                 }
 
-                $maybe = $maybe_type->getAtomicTypes();
-
-                if (count($maybe) === 1 && current($maybe) instanceof Type\Atomic\TBool) {
+                if (count($maybe_type->getAtomicTypes()) === 1
```

Comment:
`count()` could be replaced with `isSingle()`, couldn't it?

Reply:
I thought about it too, but isSingle is actually not so straightforward. For example, it will return 1 when two or more literals of the same types are present.

It's usually fine, but I'm not perfectly sure it won't create edge cases